src/BitStringAddresses/bitstring.jl

```diff
@@ -151,35 +177,35 @@ function BitString{B}(i::Union{Int128,Int64,Int32,Int16,Int8}) where {B}
     return remove_ghost_bits(BitString{B}(unsigned(i)))
 end
 function BitString{B}(i::Union{UInt64,UInt32,UInt16,UInt8}) where {B}
-    N = num_chunks(Val(B))
-    s = setindex(zero(SVector{N,UInt64}), UInt64(i), N)
-    return remove_ghost_bits(BitString{B,N}(s))
+    N, T = num_chunks(Val(B))
+    s = setindex(zero(SVector{N,T}), T(i), N)
+    return remove_ghost_bits(BitString{B,N,T}(s))
 end
 function BitString{B}(i::UInt128) where {B}
-    N = num_chunks(Val(B))
-    left = i >>> 0x40 % UInt64
-    right = i & typemax(UInt64) % UInt64
+    N, T = num_chunks(Val(B))
+    left = i >>> 0x40 % T
+    right = i & typemax(T) % T
```

Comment:
Doesn't this
```suggestion
    right = i  % T
```
do the same?

Reply:
Yeah, not sure how that got there.